src/common/options.cc

```diff
@@ -6899,6 +6899,20 @@ std::vector<Option> get_mds_options() {
     .set_default(5)
     .set_description("minimum interval between scatter lock updates"),
 
+    Option("mds_msg_client_throttle_bytes", Option::TYPE_SIZE, Option::LEVEL_ADVANCED)
+    .add_see_also("ms_dispatch_throttle_bytes")
+    .add_see_also("mds_msg_mds_throttle_bytes")
+    .set_default(64_M)
+    .set_description("number of bytes client messages can consume in memory before being throttled")
+    .set_long_description("The number of bytes client messages, for all clients, may consume in memory before the MDS begins throttling reads from the client sockets. This value should be smaller than the global throttle ms_dispatch_throttle_bytes so messages from other MDSs and Monitors may still be processed."),
+
+    Option("mds_msg_mds_throttle_bytes", Option::TYPE_SIZE, Option::LEVEL_ADVANCED)
+    .add_see_also("ms_dispatch_throttle_bytes")
+    .add_see_also("mds_msg_client_throttle_bytes")
+    .set_default(32_M)
```

Comment:
Uh, can't the MDS still dump more than this in a single migration message?

Reply:
Throttles allow overcommit as long as it's presently under the limit.

Reply:
That's a safety measure but really not something to rely on when setting limits!

Reply:
I don't see the value in raising the limit for large messages when the throttle does not prevent those messages from being read. The point is that we're only dispatching 32MB of MDS-MDS messages at a time. The real question is: what is the value in having more than 32MB of mds-mds messages in the dispatch queue? We're already processing the messages serially due to the big MDS lock.

Reply:
I'm not aware of any code in the MDS that processes MDS-MDS messages in parallel? I don't see how a deadlock could occur with the current code but perhaps a multithreaded MDS would have issues.

Reply:
I don't remember offhand if MDS messages can get delayed or reordered. I'd assume they can just because, well, why wouldn't they? Eg the lock requests presumably have to "sleep" until client capability revokes are processed, and we can still have other messages come in and process while that is happening...

Reply:
can't mds-mds messages get put on lock wait queues and such?

fwiw my inclination would be to set the client throttle to something low-ish and set the mds-mds throttle to something very big.